internal/services/ipfs_service.go

```diff
@@ -0,0 +1,83 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+const (
+	DefaultIPFSPrefix  = "ipfs://"
+	DefaultIPFSTimeout = "30s"
```

Comment:
This doesn't need to be a string
```suggestion
	DefaultIPFSTimeout = time.Second * 30
```